src/dex/miro-migrator/miro-migrator-pool.ts

```diff
@@ -0,0 +1,119 @@
+import { Interface } from '@ethersproject/abi';
+import { DeepReadonly } from 'ts-essentials';
+import { Log, Logger } from '../../types';
+import { catchParseLogError } from '../../utils';
+import { StatefulEventSubscriber } from '../../stateful-event-subscriber';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import { PoolState } from './types';
+import { uint256ToBigInt } from '../../lib/decoders';
+import MiroMigratorAbi from '../../abi/miro-migrator/MiroMigrator.abi.json';
+import ERC20ABI from '../../abi/erc20.json';
+
+export class MiroMigratorEventPool extends StatefulEventSubscriber<PoolState> {
+  handlers: {
+    [event: string]: (
+      event: any,
+      state: DeepReadonly<PoolState>,
+      log: Readonly<Log>,
+    ) => DeepReadonly<PoolState> | null;
+  } = {};
+
+  logDecoder: (log: Log) => any;
+
+  addressesSubscribed: string[];
+
+  constructor(
+    readonly parentName: string,
+    protected network: number,
+    protected dexHelper: IDexHelper,
+    logger: Logger,
+    protected migratorAddress: string,
+    protected xyzAddress: string,
+    protected transferTopic: string,
+    protected migratorInterface: Interface = new Interface(MiroMigratorAbi),
+    protected xyzInterface: Interface = new Interface(ERC20ABI),
+  ) {
+    super(parentName, 'state', dexHelper, logger);
```

Comment:
its better if subscriber has a meaningful name
```suggestion
    super(parentName, 'xyz', dexHelper, logger);
```